internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -2990,19 +2989,26 @@ func getRepoHostStatus(repoHost *appsv1.StatefulSet) *v1beta1.RepoHostStatus {
 // existing/current status for any repos in the cluster, the repository volumes
 // (i.e. PVCs) reconciled  for the cluster, and the hashes calculated for the configuration for any
 // external repositories defined for the cluster.
-func getRepoVolumeStatus(repoStatus []v1beta1.RepoStatus, repoVolumes []*corev1.PersistentVolumeClaim,
+func getRepoVolumeStatus(repoStatus []v1beta1.RepoStatus, repoVolumes map[string]*corev1.PersistentVolumeClaim,
 	configHashes map[string]string, replicaCreateRepoName string) []v1beta1.RepoStatus {
 
 	// the new repository status that will be generated and returned
 	updatedRepoStatus := []v1beta1.RepoStatus{}
 
 	// Update the repo status based on the repo volumes (PVCs) that were reconciled.  This includes
 	// updating the status for any existing repository volumes, and adding status for any new
-	// repository volumes.
-	for _, rv := range repoVolumes {
+	// repository volumes. If there was a problem with the volume when an apply was attempted,
+	// the existing status is preserved.
+	for repoName, rv := range repoVolumes {
 		newRepoVolStatus := true
-		repoName := rv.Labels[naming.LabelPGBackRestRepo]
 		for _, rs := range repoStatus {
+			// Preserve the previous status if it exists and the apply failed.
+			if rs.Name == repoName && rv == nil {
+				updatedRepoStatus = append(updatedRepoStatus, rs)
```

Comment:
❓Should we be clearing anything out of the status here?

For instanc, when `rv` is `nil` we know the PVC intent didn't apply, so should be remove any status that we set based on a value from `rv`?  It looks like this would specifically include `VolumeName` and `Bound`.

Reply:
My thinking was that, if an apply didn't take, we effectively pause updates to the status such that no status changes are made until the apply issues are sorted. Maybe I'm misunderstanding or missing something though?